### ISSUE TITLE
Strange interaction between -splitspecial and sunlights.

### ISSUE THREAD

**Issue by bstordeur** [ericwa/ericw-tools#269]
As I mentioned on Discord, getting some weird stuff with sunlights when I use -splitspecial.
Here's a test map where you can easily reproduce it, check the lighting on the left side of the building, with and without -splitspecial (looks as expected without it).
[lightextrabug.zip](https://github.com/ericwa/ericw-tools/files/3294519/lightextrabug.zip)

**Comment by hemebond** [comment]
@bstordeur Is this related to these lightmap artifacts? This is what I see when using `-splitspecial` with a `_sun` light entity; doesn't seem to affect the `worldspawn` `_sunlight`.

![image](https://github.com/ericwa/ericw-tools/assets/424218/3996088b-65e5-4b3d-bc35-16ea03aa6899)

**Comment by ericwa** [comment]
@hemebond can you upload the .map + .bsp? Also which tools version? I'm not getting those artifacts in 2.0.0-alpha7, compiling with just `-splitspecial` on qbsp:

![image](https://github.com/ericwa/ericw-tools/assets/239161/ad4ffe7d-69a9-4f6f-87d6-34eb3025dac4)

my map + bsp + lit:
[q1_light_suntexture.zip](https://github.com/ericwa/ericw-tools/files/14812868/q1_light_suntexture.zip)

That said, `-splitspecial` causing lightmapped sky is sketchy, and I'm thinking lightmapping sky should be removed. I can't think of any use case, and light.exe isn't designed for that so the results are going to be weird.

**Comment by hemebond** [comment]
@ericwa Here you go.
```
$ qbsp --version
---- qbsp / ericw-tools 2.0.0-alpha7 ----
$ light --version
---- ClearLightmapSurfaces ----
---- light / ericw-tools 2.0.0-alpha7 ----
```

![image](https://github.com/ericwa/ericw-tools/assets/424218/5eab9e47-1251-49a8-85d0-f15822393fff)
[test.zip](https://github.com/ericwa/ericw-tools/files/14813010/test.zip)

**Comment by ericwa** [comment]
Thanks, I can repro the black spots on Linux only running 2.0.0-alpha7 (Ubuntu 23.04, embree from distro packages).
Not on the windows ericw-tools binary of 2.0.0-alpha7.

Like you said it only happens with `-splitspecial`. Still investigating...

**Comment by ericwa** [comment]
> ![image](https://private-user-images.githubusercontent.com/424218/318169652-3996088b-65e5-4b3d-bc35-16ea03aa6899.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE4MzkyNjgsIm5iZiI6MTcxMTgzODk2OCwicGF0aCI6Ii80MjQyMTgvMzE4MTY5NjUyLTM5OTYwODhiLTY1ZTUtNGIzZC1iYzM1LTE2ZWEwM2FhNjg5OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMzMFQyMjQ5MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YjlmNGVjZTUyZWRkY2YzMzY0ZTVmNWZiNDY2MzgwM2JiZTc0NDUxYTJiZmMxMmRkZGMxM2ZiNTFiODQ1Y2ZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.BRkNO8FZAG9XeySgPkKwjJWlVvWz4GYU24qvp8PRiw8)

I fixed this in 21b3b696f32812b87c90d5908f8e208bc10c79e2. 

The reason `-splitspecial` was triggering the bug was just because of the fact that it chops the sky faces; I was able to repro the problem by manually chopping up a sky brush and applying different sky textures to prevent the faces from getting merged.

The underlying issue was rays missing the sky due to not setting embree to "robust" mode.

**Comment by ericwa** [comment]
I'm not able to repro any issues in 2.0.0-alpha7 with `-splitspecial` on @bstordeur 's map, my bad for not investigating this for 5 years.. It's possible it was the same issue I just fixed, or something else that got fixed in the meantime.
